infra/feast-operator/internal/controller/services/repo_config.go

```diff
@@ -35,48 +35,75 @@ func (feast *FeastServices) GetServiceFeatureStoreYamlBase64(feastType FeastServ
 }
 
 func (feast *FeastServices) getServiceFeatureStoreYaml(feastType FeastServiceType) ([]byte, error) {
-	return yaml.Marshal(feast.getServiceRepoConfig(feastType))
+	repoConfig, err := feast.getServiceRepoConfig(feastType)
+	if err != nil {
+		return nil, err
+	}
+	return yaml.Marshal(repoConfig)
 }
 
-func (feast *FeastServices) getServiceRepoConfig(feastType FeastServiceType) RepoConfig {
-	appliedSpec := feast.FeatureStore.Status.Applied
+func (feast *FeastServices) getServiceRepoConfig(feastType FeastServiceType) (RepoConfig, error) {
+	return getServiceRepoConfig(feastType, feast.FeatureStore)
+}
 
-	repoConfig := feast.getClientRepoConfig()
+func getServiceRepoConfig(feastType FeastServiceType, featureStore *feastdevv1alpha1.FeatureStore) (RepoConfig, error) {
+	appliedSpec := featureStore.Status.Applied
+
+	repoConfig := getClientRepoConfig(featureStore)
+	isLocalRegistry := IsLocalRegistry(featureStore)
 	if appliedSpec.Services != nil {
 		// Offline server has an `offline_store` section and a remote `registry`
 		if feastType == OfflineFeastType && appliedSpec.Services.OfflineStore != nil {
-			repoConfig.OfflineStore = OfflineStoreConfig{
-				Type: OfflineDaskConfigType,
+			fileType := string(OfflineDaskConfigType)
+			if appliedSpec.Services.OfflineStore.Persistence != nil && appliedSpec.Services.OfflineStore.Persistence.FilePersistence != nil {
+				fileType = appliedSpec.Services.OfflineStore.Persistence.FilePersistence.Type
+			}
+
+			repoConfig.OfflineStore = OfflineStoreConfig{}
+			var err error
+			repoConfig.OfflineStore.Type, err = ParseOfflineConfigType(fileType)
```

Comment:
```suggestion
			repoConfig.OfflineStore.Type, err := ParseOfflineConfigType(fileType)
```

Reply:
we shouldn't require a `ParseOfflineConfigType`. we should be able to just set the CRD to use the `OfflineConfigType` directly.